### PR TITLE
fix shared ptr

### PR DESCRIPTION
**Pull request by chchatte92**
### Briefly, what does this PR introduce?
This PR introduces the fixes needed to pixel gap cut test following the new `gsl::not_null` 

### What kind of change does this PR introduce?
- [X] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [X ] Changes have been communicated to collaborators
     I am discussing the changes with @wdconinc and @c-dilks 
### Does this PR change default behavior?
          The testing script currently crashes.

### CODE REVIEW

**Comment by chchatte92** [comment]
This line seems problematic, the print statements after is not getting executed.
`auto pos_local = geo.GetSensorLocalPosition(cellID, pos_global);`
https://github.com/eic/drich-dev/blob/a328bf27d1bbd040460a55a61a30b7986b5b7385/src/test_pixel_gap_cuts.cpp#L77

File: src/test_pixel_gap_cuts.cpp [hunk 1]
```diff
@@ -43,11 +44,12 @@ int main(int argc, char** argv) {
     return 1;
   }
   auto compactFile = DETECTOR_PATH + "/" + DETECTOR_CONFIG + ".xml";
-  auto det = &dd4hep::Detector::getInstance();
-  det->fromXML(compactFile);
+  dd4hep::Detector & det = dd4hep::Detector::getInstance();
```
Comment (wdconinc) [inline src/test_pixel_gap_cuts.cpp:47]: This is a deprecated interface, https://github.com/AIDASoft/DD4hep/blob/1962f7d1ace4484efb8886d66a6759a83c5107a3/DDCore/include/DD4hep/Detector.h#L77-L82. Ownership semantics better expressed with `make_unique`, https://github.com/AIDASoft/DD4hep/blob/1962f7d1ace4484efb8886d66a6759a83c5107a3/DDCore/include/DD4hep/Detector.h#L77-L82.

File: src/test_pixel_gap_cuts.cpp [hunk 2]
```diff
@@ -43,11 +44,12 @@ int main(int argc, char** argv) {
...
+  det.fromXML(compactFile);
+  dd4hep::rec::CellIDPositionConverter cellid_converter(det);
 
   // ReadoutGeo
-  richgeo::ReadoutGeo geo("DRICH", det, logger);
+  richgeo::ReadoutGeo geo("DRICH", gsl::not_null{ &det}, gsl::not_null{ &cellid_converter}, logger);
```
Comment (wdconinc) [inline src/test_pixel_gap_cuts.cpp:52]: gsl::not_null unneeded for address of object: always not_null.
```suggestion
  richgeo::ReadoutGeo geo("DRICH", &det, &cellid_converter, logger);
```

File: src/test_pixel_gap_cuts.cpp (hunk 1 above)
Comment (chchatte92) [reply src/test_pixel_gap_cuts.cpp:47]: The both links are pointing to the same line. can you please point where can I look for the better expression with `make_unique`?

Comment (wdconinc) [reply src/test_pixel_gap_cuts.cpp:47]: Further in the same file; https://github.com/AIDASoft/DD4hep/blob/1962f7d1ace4484efb8886d66a6759a83c5107a3/DDCore/include/DD4hep/Detector.h#L349

Comment (chchatte92) [reply src/test_pixel_gap_cuts.cpp:47]: Thanks a lot. But sorry again another stupid question.
This make_unique method will return me of a type unique_ptr<dd4hep::Detector>, now in order to access
ReadoutGeo, I need a CellIDPoisitionConverter which accepts just dd4hep::Detector. How can use it then?

File: src/test_pixel_gap_cuts.cpp [hunk 3]
```diff
@@ -61,7 +63,7 @@ int main(int argc, char** argv) {
   for(unsigned e=0; e<reader.getEntries(tree_name); e++) {
     logger->trace("EVENT {}", e);
     auto frame = podio::Frame(reader.readNextEntry(tree_name));
-
+    fmt::print("here\n");
```
Comment (c-dilks) [inline src/test_pixel_gap_cuts.cpp:66]:
```suggestion
```

**Comment by c-dilks** [comment]
Seems to be working now.

File: src/test_pixel_gap_cuts.cpp (hunk 1 above)
Comment (c-dilks) [reply src/test_pixel_gap_cuts.cpp:47]: > This is a deprecated interface

Meh, let's let it break when it breaks. This pixel gap test should be moved to the benchmarks, if access to the EICrecon services could be provided.

**Comment by chchatte92** [comment]
LoL. I am not sure how though!!

**Comment by chchatte92** [comment]
I found the building works perfectly well. But, when executed I could not generate the png file.  On a deeper look inside EICRecon I find that when the method GetLocalPosition is trying to find the VolumeManagerContext of the cellID, it does not work and I have understood that is what actually the crash (execution) talks about:
`
VolumeManager    ERROR lookupContext: Failed to search Volume context 00040007001D7D78 [Unknown identifier]
terminate` called after throwing an instance of 'std::runtime_error'
  what():  VolumeManager: lookupContext: Failed to search Volume context 00040007001D7D78 [Unknown identifier]
Aborted (core dumped)
`

**Comment by chchatte92** [comment]
Please ignore my all stupid comments /concerns before....
https://github.com/eic/drich-dev/pull/111#issuecomment-1751682004
It is definitely working!!
I was plugging in ana.edm4hep.root, and @c-dilks you know, for very obvious reasons the test did not find any sim hits!!!!
 With a meaningful input here it is what it produces...
image

**Comment by c-dilks** [comment]
If this is ready for review+merge, please mark it ready.

**Review by c-dilks — APPROVED** [review]

**Pull request merged**